packages/storage/src/cookies.ts

```diff
@@ -89,45 +93,16 @@ try {
 export const cookieStorage: StorageWithOptions<CookieOptions> = addClearMethod({
   _read: isServer
     ? (options?: CookieOptions) => {
-        const eventOrRequest = options?.getRequest?.() || useRequest();
-        const request =
-          eventOrRequest && ("request" in eventOrRequest ? eventOrRequest.request : eventOrRequest);
-        let result = "";
-        if (eventOrRequest.responseHeaders) {
-          // Check if we really got a pageEvent
-          const responseHeaders = eventOrRequest.responseHeaders as Headers;
-          result +=
-            responseHeaders
-              .get("Set-Cookie")
-              ?.split(",")
-              .map(cookie => !cookie.match(/\\w*\\s*=\\s*[^;]+/))
-              .join(";") ?? "";
-        }
-        return `${result};${request?.headers?.get("Cookie") ?? ""}`; // because first cookie will be preferred we don't have to worry about duplicates
-      }
+      const headers = options?.getRequestHeaders?.() || getRequestHeaders();
+      return headers?.get("Cookie") ?? "";
+    }
     : () => document.cookie,
   _write: isServer
-    ? (key: string, value: string, options?: CookieOptions) => {
-        if (options?.setCookie) {
-          options?.setCookie?.(key, value, options);
-          return;
-        }
-        const pageEvent: PageEvent = options?.getRequest?.() || useRequest();
-        if (!pageEvent.responseHeaders)
-          // Check if we really got a pageEvent
-          return;
-        const responseHeaders = pageEvent.responseHeaders as Headers;
-        const cookies =
-          responseHeaders
-            .get("Set-Cookie")
-            ?.split(",")
-            .filter(cookie => !cookie.match(`\\s*${key}\\s*=`)) ?? [];
-        cookies.push(`${key}=${value}${serializeCookieOptions(options)}`);
-        responseHeaders.set("Set-Cookie", cookies.join(","));
-      }
+    ? (key: string, value: string, options?: CookieOptions) =>
+      (options?.setCookie ?? setCookie)?.(key, value, options as CookieProperties)
     : (key: string, value: string, options?: CookieOptions) => {
-        document.cookie = `${key}=${value}${serializeCookieOptions(options)}`;
-      },
+      document.cookie = `${key}=${value}${serializeCookieOptions(options)}`;
+    },
   getItem: (key: string, options?: CookieOptions) =>
     deserializeCookieOptions(cookieStorage._read(options), key),
```

Comment:
This `window.dispatchEvent` call should be unnecessary when we already have this new shiny sync or isn't it?

Reply:
That is indeed correct; it never worked as expected in any case, since it is not propagated to other tabs.